aloha-core/src/main/scala/com/eharmony/aloha/audit/impl/EitherAuditor.scala

```diff
@@ -0,0 +1,175 @@
+package com.eharmony.aloha.audit.impl
+
+import com.eharmony.aloha.audit.MorphableAuditor
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.reflect.RefInfo
+
+import scala.collection.breakOut
+
+/**
+  * An [[EitherAuditor]] encodes the idea of success or failure.  In the case of failure,
+  * (meaning a model could not produce a prediction) diagnostic information is encoded
+  * in the form of an [[EitherAuditorError]].  No diagnostic information is emitted when
+  * a "''successful''" prediction is produced, even if compromises were made in order to
+  * produce the prediction or problems like missing data were encountered.
+  *
+  * To construct an [[EitherAuditor]], use the `apply` factory methods in the companion
+  * object.
+  *
+  * Created by ryan.deak on 10/2/17.
+  * @tparam A the type of value returned in the success case.
+  */
+sealed abstract class EitherAuditor[A]
+extends MorphableAuditor[Either[EitherAuditorError, _], A, Either[EitherAuditorError, A]] {
+
+  /**
+    * Return an EitherAuditor for success type `M`.
+    * @tparam M type of values returned on success
+    * @return a new auditor. An EitherAuditor can be returned for any type `M`.
+    */
+  override def changeType[M: RefInfo]: Some[EitherAuditor[M]] = Some(changeTo[M])
```

Comment:
Just looking at the signature of this function it looks wonky.  Why bother with a `Some()` if failure is impossible.

Reply:
The contract of `MorphableAuditor` is 

```scala
def changeType[M: RefInfo]: Option[ /* Auditor of similar type parametrized by M */ ]
```

It's possible that an auditor can't produce a similar auditor based on type `M`.  So we return an `Option`.  Because functions can return subtypes and `Some` is an `Option`, we can explicitly return `Some` since functions are covariant in their output type.